.github/workflows/producer_dbt.yml

```diff
@@ -0,0 +1,127 @@
+        description: "release tag of OpenLineage to use"
+        type: string
+        default: "1.23.0"
+      get-latest-snapshots:
+        description: "Should the artifact be downloaded from maven repo or circleci"
+        type: string
+        default: "false"
+
+jobs:
+  run-dbt-tests:
+    runs-on: ubuntu-latest
+    
+    services:
+      postgres:
+        image: postgres:15-alpine
+        env:
+          POSTGRES_USER: testuser
+          POSTGRES_PASSWORD: testpass
+          POSTGRES_DB: dbt_test
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd "pg_isready -U testuser -d dbt_test"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Initialize tests
+        id: init
+        run: |
+          scenarios=$(./scripts/get_valid_test_scenarios.sh "producer/dbt/scenarios/" ${{ inputs.dbt_release }} ${{ inputs.ol_release }} )
+          if [[ "$scenarios" != "" ]]; then
+            echo "scenarios=$scenarios" >> $GITHUB_OUTPUT
+            echo "Found scenarios: $scenarios"
+          else
+            echo "No valid scenarios found for dbt ${{ inputs.dbt_release }} and OL ${{ inputs.ol_release }}"
+          fi
+
+      - name: Set up Python 3.12
+        if: ${{ steps.init.outputs.scenarios }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install dbt dependencies
+        if: ${{ steps.init.outputs.scenarios }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install dbt-core==${{ inputs.dbt_release }}
+          pip install dbt-postgres
+          pip install openlineage-dbt==${{ inputs.ol_release }}
+          pip install -r producer/dbt/test_runner/requirements.txt
+
+      - name: Set producer output event dir
+        if: ${{ steps.init.outputs.scenarios }}
+        id: set-producer-output
+        run: | 
+          echo "event_dir=/tmp/dbt-events-$(date +%s%3N)" >> $GITHUB_OUTPUT
+
+      - name: Run dbt scenarios and create OL events
+        if: ${{ steps.init.outputs.scenarios }}
+        id: run-producer
+        continue-on-error: true
+        run: |          
+          set -e
+          IFS=';' read -ra scenarios <<< "${{ steps.init.outputs.scenarios }}"
+          
+          for scenario in "${scenarios[@]}"
+          do
+              echo "Running dbt scenario: $scenario"
+
+              if ! python3 producer/dbt/test_runner/cli.py run-scenario \
+                --scenario "$scenario" \
+                --output-dir "${{ steps.set-producer-output.outputs.event_dir }}"
+              then
+                echo "Error: dbt scenario failed: $scenario"
+                exit 1
+              fi
+
+              echo "Finished running scenario: $scenario"
+          done
+
+          echo "Finished running all scenarios"
+
+      - name: Validation
```

Comment:
I see that you use validation script but there is also validation in the test scenario file, is there some reason for it?